scripts/lib/CIME/XML/machines.py

```diff
@@ -175,13 +175,13 @@ def _probe_machine_name_one_guess(self, nametomatch):
                 logger.debug("machine regex string is " + regex_str)
                 # an environment variable can be used
                 if regex_str.startswith("$ENV"):
-                    machine = self.get_resolved_value(regex_str, allow_unresolved_envvars=True)
-                    if not machine.startswith("$ENV"):
+                    machine_value = self.get_resolved_value(regex_str, allow_unresolved_envvars=True)
+                    if not machine_value.startswith("$ENV"):
                         try:
-                            match,machine = machine.split(":")
+                            match, this_machine = machine_value.split(":")
```

Comment:
We don't absolutely need a new variable here (`this_machine`): we could reuse `machine_value` like the old code reused `machine`. However, I have introduced a new variable for clarity, because I find it confusing when code uses the same variable for multiple purposes.